src/realm/obj.cpp

```diff
@@ -2070,21 +2070,20 @@ struct EmbeddedObjectLinkMigrator : public LinkTranslator {
 void Obj::handle_multiple_backlinks_during_schema_migration()
 {
     REALM_ASSERT(!m_table->get_primary_key_column());
-    auto copy_links = [this](ColKey col) {
-        auto embedded_obj_tracker = std::make_shared<converters::EmbeddedObjectConverter>();
```

Comment:
EmbeddedObjectConverter is always created by the caller and outlives the things which use it, so it can just be stack allocated rather than stored in a shared_ptr.